File: insight-brain-frontend/src/main/frontend/owner.manager/source.control/source.control.tile.js
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
import { selectIsSourceControlForSourceTileSupported } from 'MainRoot/productFeatures/productFeaturesSelectors';
import { selectSelectedOwnerName } from 'MainRoot/OrgsAndPolicies/orgsAndPoliciesSelectors';

import template from './source.control.tile.html';
import {
  selectSourceControl,
  selectEffectiveProvider,
  selectItemSubText,
  selectItemText,
  selectLoading,
  selectLoadError,
} from 'MainRoot/OrgsAndPolicies/sourceControlSelectors';
import {
  selectIsApplication,
  selectIsOrganization,
  selectIsRootOrganization,
} from 'MainRoot/reduxUiRouter/routerSelectors';

export default {
  template: template,
  controllerAs: 'vm',
  controller: SourceControlTileController,
};

function SourceControlTileController($scope, SameOwnerStateNavigationService, $ngRedux) {
  const vm = this;

  vm.editSourceControl = editSourceControl;

  vm.unsubscribe = $ngRedux.connect(mapStateToThis)(vm);

  $scope.$on('$destroy', function () {
    vm.unsubscribe();
  });

  function editSourceControl() {
    SameOwnerStateNavigationService.goEdit('edit-source-control');
  }
}
const mapStateToThis = (state) => ({
  isSourceControlSupported: selectIsSourceControlForSourceTileSupported(state),
  ownerName: selectSelectedOwnerName(state),
  sourceControl: selectSourceControl(state),
  isApp: selectIsApplication(state),
  isOrg: selectIsOrganization(state),
  isRootOrg: selectIsRootOrganization(state),
  effectiveProvider: selectEffectiveProvider(state),
  itemText: selectItemText(state),
  itemSubText: selectItemSubText(state),
  loading: selectLoading(state),
  error: selectLoadError(state),
});

SourceControlTileController.$inject = ['$scope', 'SameOwnerStateNavigationService', '$ngRedux'];

File: insight-brain-testing/insight-brain-functional-test-common/src/main/java/com/sonatype/clm/testing/functional/elements/SourceControlTile.java
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
package com.sonatype.clm.testing.functional.elements;

import com.codeborne.selenide.ElementsCollection;
import com.codeborne.selenide.SelenideElement;

public class SourceControlTile
    extends OwnerTile
{
  private static final String SOURCE_CONTROL_OWNER_ELEMENT_ID = "#owner-pill-source-control";

  public SourceControlTile() {
    super(SOURCE_CONTROL_OWNER_ELEMENT_ID);
  }

  public ElementsCollection rows() {
    return children(".nx-list__item");
  }

  public SelenideElement itemText() {
    return child(".nx-list__text");
  }

  public SelenideElement itemSubText() {
    return child(".nx-list__subtext");
  }

  public SelenideElement content() {
    return child(".nx-tile-content");
  }

  public SelenideElement sourceControlLink() {
    return child(".nx-list__link");
  }
}
